### PR TITLE
run-986 - new imagePullPolicy flag

### PR DESCRIPTION
**Pull request by galbarnissan**

### CODE REVIEW

File: cmd/job/submit/image_pull_policy_handler.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package submit
+
+import "fmt"
+
+func handleImagePullPolicy(submitArgs *submitArgs) (err error) {
+	switch submitArgs.ImagePullPolicy {
+	case "always", "ifNotPresent", "never":
```
Comment (yodarshafrir1) [inline cmd/job/submit/image_pull_policy_handler.go:7]: Shouldn't be `Always`, `IfNotPresent` and `Never`?

File: cmd/job/submit/image_pull_policy_handler.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+		if submitArgs.LocalImage != nil && *submitArgs.LocalImage {
+			submitArgs.ImagePullPolicy = "never"
+		}
+		if submitArgs.AlwaysPullImage != nil && *submitArgs.AlwaysPullImage {
```
Comment (yodarshafrir1) [inline cmd/job/submit/image_pull_policy_handler.go:11]: The default value is `Always` no? so this part has no meaning right?

File: cmd/job/submit/image_pulll_policy_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
+package submit
```
Comment (yodarshafrir1) [inline cmd/job/submit/image_pulll_policy_handler_test.go:1]: Nice 🥇

**Review by yodarshafrir1 — CHANGES_REQUESTED** [review]
Looking good,
Some minor fixes, please see my notes

File: cmd/job/submit/image_pull_policy_handler.go (hunk 2 above)
Comment (galbarnissan) [reply cmd/job/submit/image_pull_policy_handler.go:11]: yes, this can be removed.

File: cmd/job/submit/submit.go [hunk 4]
```diff
@@ -207,12 +208,15 @@ func (submitArgs *submitArgs) addCommonFlags(fbg flags.FlagsByGroups) {
...
+	flagSet.StringVar(&(submitArgs.ImagePullPolicy), "image-pull-policy", "Always", "the policy to pull the image, and the default policy is always")
 	flags.AddBoolNullableFlag(flagSet, &(submitArgs.AlwaysPullImage), "always-pull-image", "", "Always pull latest version of the image.")
+	flagSet.MarkDeprecated("always-pull-image", "please use 'image-pull-policy=always' instead.")
 	flagSet.StringArrayVar(&(submitArgs.Args), "args", []string{}, "Arguments to pass to the command run on container start. Use together with --command.")
 	flagSet.StringArrayVarP(&(submitArgs.EnvironmentVariable), "environment", "e", []string{}, "Set environment variables in the container.")
 	flagSet.StringVarP(&(submitArgs.Image), "image", "i", "", "Container image to use when creating the job.")
 	flagSet.StringArrayVar(&(submitArgs.Command), "command", []string{}, "Run this command on container start. Use together with --args.")
 	flags.AddBoolNullableFlag(flagSet, &submitArgs.LocalImage, "local-image", "", "Use an image stored locally on the machine running the job.")
+	flagSet.MarkDeprecated("local-image", "please use 'image-pull-policy=never' instead.")
```
Comment (yodarshafrir1) [inline cmd/job/submit/submit.go:219]: 'never' -> `Never`

File: cmd/job/submit/submit.go [hunk 5]
```diff
@@ -207,12 +208,15 @@ func (submitArgs *submitArgs) addCommonFlags(fbg flags.FlagsByGroups) {
 	flagSet.MarkHidden("dry-run")
 
 	flagSet = fbg.GetOrAddFlagSet(ContainerDefinitionFlagGroup)
+	flagSet.StringVar(&(submitArgs.ImagePullPolicy), "image-pull-policy", "Always", "the policy to pull the image, and the default policy is always")
 	flags.AddBoolNullableFlag(flagSet, &(submitArgs.AlwaysPullImage), "always-pull-image", "", "Always pull latest version of the image.")
+	flagSet.MarkDeprecated("always-pull-image", "please use 'image-pull-policy=always' instead.")
```
Comment (yodarshafrir1) [inline cmd/job/submit/submit.go:213]: `always` -> `Always`

File: cmd/job/submit/submit.go [hunk 6]
```diff
@@ -207,12 +208,15 @@ func (submitArgs *submitArgs) addCommonFlags(fbg flags.FlagsByGroups) {
 	flagSet.MarkHidden("dry-run")
 
 	flagSet = fbg.GetOrAddFlagSet(ContainerDefinitionFlagGroup)
+	flagSet.StringVar(&(submitArgs.ImagePullPolicy), "image-pull-policy", "Always", "the policy to pull the image, and the default policy is always")
```
Comment (yodarshafrir1) [inline cmd/job/submit/submit.go:211]: `the policy of image pull, set to "Always" by default.`

**Review by yodarshafrir1 — CHANGES_REQUESTED** [review]
last fixes

**Review by yodarshafrir1 — APPROVED** [review]
💯

**Pull request merged**